File: clients/ev3-java/src/main/java/ch/zhaw/ev3/sensors/Sonic.java
package ch.zhaw.ev3.sensors;

import ch.zhaw.ev3api.api.SensorApi;
import ch.zhaw.ev3api.invoker.ApiException;

public class Sonic extends Sensor {
    public enum Modes {
        DIST_CM("US-DIST-CM"),
        DIST_IN("US-DIST-IN"),
        LISTEN("US-LISTEN"),
        SI_CM("US-SI-CM"),
        SI_IN("US-SI-IN"),
        DC_CM("US-DC-CM"),
        DC_IN("US-DC-IN");

        private final String mode;

        Modes(String mode) {
            this.mode = mode;
        }


        @Override
        public String toString() {
            return mode;
        }
    }

    public Sonic(SensorApi sensorApi) throws ApiException {
        super(Drivers.SONIC, sensorApi);
    }

    public int getDistance() throws ApiException {
        setMode(Modes.DIST_CM.mode);
        return Integer.parseInt(getValues().get(0));
    }
}

File: clients/ev3-python/ev3/sensors/Sonic.py
from enum import Enum
from ev3.sensors.Sensor import Sensor
from ev3api.api.sensor_api import SensorApi


class Sonic(Sensor):
    class MODES(Enum):
        DIST_CM = "US-DIST-CM"
        DIST_IN = "US-DIST-IN"
        LISTEN = "US-LISTEN"
        SI_CM = "US-SI-CM"
        SI_IN = "US-SI-IN"
        DC_CM = "US-DC-CM"
        DC_IN = "US-DC-IN"

    def __init__(self, sensor_api: SensorApi):
        super(Sonic, self).__init__(sensor_api, Sensor.Drivers.SONIC)

    def get_distance(self) -> int:
        self.set_mode(self.MODES.DIST_CM.value)
        return int(self.get_values()[0])
